src/components/ServiceList.jsx

```diff
@@ -30,32 +54,68 @@ const ServiceList = () => {
     );
   }
 
-  return isLoading ? (
-    <p>Loading Baltimore County services...</p>
-  ) : (
-    <div>
-      <div style={flex}>
-        <i
-          className="fas fa-star dg_popular_star"
-          aria-hidden="true"
-          style={MostPopularServiceIconStyles}
-        ></i>
-        <p style={legendText}> - Indicates a Most Popular Service</p>
-      </div>
-      <div className="row">
-        <FilterList
-          items={serviceItems}
-          renderItem={props => (
-            <div
-              key={props.name.replace(/\s/, "-")}
-              className="d-flex col-lg-3 col-md-6 col-sm-6"
-            >
-              <ServiceIconLink {...props} />
-            </div>
-          )}
-        />
-      </div>
-    </div>
+  const hasFilteredResults = !(isFiltering && filteredItems.length === 0);
+
+  return (
+    <React.Fragment>
+      {isLoading ? (
+        <p>Loading Baltimore County services...</p>
+      ) : (
+        <div className="row">
+          <div className="col-md-3 col-xs-12">
+            <PopularityFilterCollapse
+              header="Filter"
+              id="Popular-filter"
+              onChange={handleIsPopularFilterChange}
+              checked={isMostPopular}
+            />
+          </div>
+          <div className="col-md-9 col-xs-12">
+            <TextInput
+              id="full-name"
+              label="Search for services"
+              placeholder="Search for services and more..."
+              onChange={handleTextInputFilterChange}
+            />
+
+            {hasFilteredResults ? (
+              <div>
+                <ListLegend
+                  icon="fas fa-star"
+                  text="- Indicates a Most Popular Service"
+                />
+
+                <div className="row">
+                  <FilterList
+                    items={
+                      filteredItems.length > 0 ? filteredItems : serviceItems
+                    }
+                    renderItem={props => (
+                      <div
+                        key={props.name.replace(/\s/, "-")}
+                        className="d-flex col-lg-3 col-md-6 col-sm-6"
+                      >
+                        <ServiceIconLink {...props} checked={isMostPopular} />
+                      </div>
+                    )}
+                  />
+                </div>
+                <ListCounter
```

Comment:
I suspect we are going to need to move this in the same line as the legend, but we can let that be discovered during testing.